internal/home/service.go

```diff
@@ -681,3 +687,64 @@ rc_bg=YES
 
 rc_cmd $1
 `
+
+// secureBinary is used before service.Install(). This function protects AdGuardHome from
+// privilege escalation vulnerabilities caused by writable files
+func secureBinary() error {
+	switch runtime.GOOS {
```

Comment:
In new code, we prefer to use build tags for OS-specific pieces of functionality.  See package `aghos`.  In fact, this function should probably be there.